lib/es6-promise/enumerator.js

```diff
@@ -14,42 +14,37 @@ import {
 } from './-internal';
 
 export function makeSettledResult(state, position, value) {
-  if (state === FULFILLED) {
-    return {
-      state: 'fulfilled',
-      value: value
-    };
-  } else {
-    return {
-      state: 'rejected',
-      reason: value
-    };
-  }
+  return {
+    state: (state === FULFILLED) ? 'fulfilled' : 'rejected',
+    value: value
+  };
 }
 
 function Enumerator(Constructor, input, abortOnReject, label) {
-  this._instanceConstructor = Constructor;
-  this.promise = new Constructor(noop, label);
-  this._abortOnReject = abortOnReject;
+  var self = this;
```

Comment:
once last request, `self` -> something more meaningful like `enumerator`, as `self` has meaning in the browser and web workers.

Also you don't need to open a new PR, rather you can just force push to your existing branch that is the source of this update and it will update accordingly